t/50chunked-encoding-proxying.t

```diff
@@ -0,0 +1,82 @@
+use strict;
+use warnings;
+use Net::EmptyPort qw(check_port empty_port);
+use Test::More;
+use t::Util;
+use Time::HiRes qw(usleep);
+
+
+plan skip_all => 'nc not found'
+    unless prog_exists('nc');
+
+my $upstream_port = empty_port();
+$| = 1;
+my $socket = new IO::Socket::INET (
+    LocalHost => '127.0.0.1',
+    LocalPort => $upstream_port,
+    Proto => 'tcp',
+    Listen => 1,
+    Reuse => 1
+);
+die "cannot create socket $!\n" unless $socket;
+
+check_port($upstream_port) or die "can't connect to server socket";
+# accent and close check_port's connection
+my $client_socket = $socket->accept();
+close($client_socket);
+
+my $server = spawn_h2o(<< "EOT");
+hosts:
+  default:
+    paths:
+      "/":
+        proxy.reverse.url: http://127.0.0.1:$upstream_port
+EOT
+
+sub doit {
+    my $chunk = shift;
+    my $data = shift;
+    my $stream_window_bits = shift;
+    my $defer_close = shift;
+    my $expect_rst_stream = shift;
+    open(NGHTTP, "nghttp -w $stream_window_bits -v http://127.0.0.1:$server->{'port'}/ -H 'host: host.example.com' 2>&1 |");
```

Comment:
You might want to:
* use a scalar for holding the filehandle, i.e. `open my $nghttp, ...`
* use three-argument open, i.e. `open my $nghttp, "-|", "nghttp...`